### DISCUSSION TITLE
OsgEarth and IMGUI in a personal project

### DISCUSSION THREAD

**Discussion by Mattc2021** [pelicanmapping/osgearth#2205] [Support Q&A]
So I was trying to get OSGEarth to be implemented into a project using vcpkg and when I tried to test to see if the imgui functionality was working with it by taking the src/application/osgearth_imgui code and building it it came back with multiple linker errors. I was curious if anyone has run into this problem or if anyone knew of a solution. I've tried messing with my vcpkg.json and adding osgearth[tools]:x64-windows which didn't seem to work. I've also just tried grabbing the osgearth/imgui folder and dropping that in and that didn't work. Any help would be greatly appreciated!

The packages that I have being installed through vcpkg are 
```
  "dependencies": [
    {
      "name": "osg",
      "features": [
        "plugins"
      ]
    },
    "osg",
    {
      "name": "osgearth",
      "features": [
        "tools"
      ]
    },
    "imgui",
    "glew"
  ],
```

**Comment by jasonbeverage** [comment]
What kind of errors are you seeing?  Does osgearth_imgui build properly and work when installed via vcpkg for you?

**Reply by Mattc2021** [reply]
![image](https://user-images.githubusercontent.com/46578874/211910157-181c2e77-2f6d-44d4-9e49-f0ebbdd6a366.png)

These are the errors I am currently seeing. Unfortunately I've never gotten the imgui portion to build properly. Everything else 99% of the time works well except for this integration.

**Reply by jepessen** [reply]
Those are linker errors. you need to link osgearth to your project. You need to include the package in the CMakeLists.txt:
```
find_package (osgEarth CONFIG REQUIRED)
```

And then add the link target:
```
target_link_libraries (
  ${PROJECT_NAME}
  osgEarth
)
```

If you're configuring the Visual Studio solution instead you need to add the osgearth `.lib` files in the configuration of the project.
